source/async_stack.cpp

```diff
@@ -38,11 +41,52 @@ extern "C" {
 // Initialise to some value that will be interpreted as an invalid key.
 inline pthread_key_t folly_async_stack_root_tls_key = 0xFFFF'FFFFu;
 }
-
-#endif  // FOLLY_ASYNC_STACK_ROOT_USE_PTHREAD
+#else 
+#   include <vector>
+#endif  //UNIFEX_ASYNC_STACK_ROOT_USE_PTHREAD
 
 namespace unifex {
 
+#if UNIFEX_ASYNC_STACK_ROOT_USE_PTHREAD == 0
+
+struct AsyncStackRootHolderList {
+  std::vector<void*> asyncStackRootHolders_;
+  std::mutex mutex_;
+
+  AsyncStackRootHolderList() {
+    // reserve some space to avoid reallocations
+    asyncStackRootHolders_.reserve(300);
```

Comment:
This seems like a lot; the vector should only have O(#threads) elements in it…

Reply:
Yeah it's definitely an overestimate. From some quick searching, I got an estimate that some apps peak between 200-500 threads

Reply:
I think it's reasonable to set this to something like 16 or 32.  At 32 this will eliminate the first 6 or so allocations for resizing the vector, to get to 500 would only require 4 more allocations and I think this is OK.